### PR TITLE
simplify description of storages; add Heat and Hydrogen options to storage

### PR DESCRIPTION
**Pull request by sandrinecharousset**
This PR removes the energy storage system level in description of variables used for storages; This level was creating unconsistencies for the hydro units description as well as unnecessary complexity.
Now variables used for Hydro reservoir and Hydro Pued storage are defined separately as this technology is behaving differently (in particular it is creating energy , and also Hydro|Reservoir and Hydro|Pumped Storage are already included in Electricity_Input whch created problems in the variable description
This PR also takes what was in PR #288  by @amosschle and PR #256 by @erikfilias which I suggest to close

### CODE REVIEW

**Comment by erikfilias** [comment]
> This PR removes the energy storage system level in description of variables used for storages; This level was creating unconsistencies for the hydro units description as well as unnecessary complexity. Now variables used for Hydro reservoir and Hydro Pued storage are defined separately as this technology is behaving differently (in particular it is creating energy , and also Hydro|Reservoir and Hydro|Pumped Storage are already included in Electricity_Input whch created problems in the variable description This PR also takes what was in PR #288 by @amosschle and PR #256 by @erikfilias which I suggest to close

close #256

**Comment by sandrinecharousset** [comment]
@erikfilias I could not include Hydrogen in tag_storage.yaml as this creates duplicate variables because Hydrogen is already used in Electricity Input (same than for hydro) I believe we are at the limit of what we can do with those tags and I suggest that as I did for Hydro, you add the variables that you need for Hydrogen storage without using the tag list. I couldn't do it because I cannot figure how you are doing it as the benefit of storing electricity into hydrogen is for me not clear at all..... and I thought your case study was focused on electricity . Does it mean what you are storing is Hydrogen as a fuel for Gas power plant using blending of gas and hydrogen?

File: definitions/variable/technology/power-plant.yaml [hunk 1]
```diff
@@ -127,40 +126,69 @@
...
     unit: GWh
-- Maximum Charge|Electricity|{Electricity Storage Type}:
-    description: Maximum charge when storing energy for a {Electricity Storage Type} unit
+- Inflows|Electricity|Hydro|Pumped Storage:
+    description: Inflows into Pumped storage, expressed in energy
+    unit: GWh
+- Outflows|Electricity|{Storage Type}:
+    description: Outflows from a storage of a {Storage Type}, expressed in energy to cover a demand outside of the electricity system
+    unit: GWh
+- Maximum Charge|Electricity|{Storage Type}:
```
Comment (erikfilias) [inline definitions/variable/technology/power-plant.yaml:175]:
```suggestion
- Outflows|Electricity|Hydrogen:
    description: Outflows from a storage of a hydrogen unit, expressed in energy to cover a demand outside of the electricity system
    unit: GWh
- Maximum Charge|Electricity|{Storage Type}:
```

File: definitions/variable/technology/power-plant.yaml [hunk 2]
```diff
@@ -127,40 +126,69 @@
...
+- Seasonality|Electricity|{Storage Type}:
     description: seasonality (daily, weekly, monthly, etc.) of {Electricity Storage Type}
     unit: [daily,weekly,monthly]
-- Roundtrip Efficiency|Electricity|{Electricity Storage Type}:
+- Roundtrip Efficiency|Electricity|{Storage Type}:
     description: Efficiency when charging energy from the grid to the system
     unit: '-'
+- Roundtrip Efficiency|Electricity|Hydro|Pumped Storage:
+    description: Efficiency when charging energy from the grid to the pumped storage
+    unit: '-'
```
Comment (erikfilias) [inline definitions/variable/technology/power-plant.yaml:194]:
```suggestion
    unit: '-'
- Roundtrip Efficiency|Electricity|Hydrogen:
    description: Efficiency when charging energy from the grid to the hydrogen storage
    unit: '-'
```

**Review by erikfilias — APPROVED** [review]

**Comment by sandrinecharousset** [comment]
> > @erikfilias I could not include Hydrogen in tag_storage.yaml as this creates duplicate variables because Hydrogen is already used in Electricity Input (same than for hydro) I believe we are at the limit of what we can do with those tags and I suggest that as I did for Hydro, you add the variables that you need for Hydrogen storage without using the tag list. I couldn't do it because I cannot figure how you are doing it as the benefit of storing electricity into hydrogen is for me not clear at all..... and I thought your case study was focused on electricity . Does it mean what you are storing is Hydrogen as a fuel for Gas power plant using blending of gas and hydrogen? In that case I have no idea which variables you may need. Regarding the hydro I did not create all the possible variables as some are nonsense ....
> 
> Thank you, Sandrine. I just suggested (in this PR) the variables related to hydrogen as storage for electricity, which we need in the CS3.

Thank you @erikfilias , I approved all + changed units : % is not allowed, mst be "-"

**Comment by sandrinecharousset** [comment]
@danielhuppmann this PR can be merged now I think. Erik has approved and included Hydrogen variables; I have included Heat as requested by Amos and closed #288

**Comment by sandrinecharousset** [comment]
@felixfrischmuth , please review this PR as it includes adding electric vehicles categories within tag_storage.yaml, which I need for uploading results of CS4

**Review by felixfrischmuth — APPROVED** [review]
I am not sure if you should include HEVs as a storage type here. I would leave them out here.     
- Electric Vehicle|Hybrid:
      description: Hybrid electric vehicle (HEV), including both an electric motor and a combustion engine running on gas, that cannot be plugeed in to charge

**Comment by sandrinecharousset** [comment]
> I am not sure if you should include HEVs as a storage type here. I would leave them out here.
> 
> * Electric Vehicle|Hybrid:
>   description: Hybrid electric vehicle (HEV), including both an electric motor and a combustion engine running on gas, that cannot be plugeed in to charge

Yes you are completely right, will remove @felixfrischmuth

**Comment by danielhuppmann** [comment]
closing and opening to restart the tests

**Review by danielhuppmann — APPROVED** [review]
Thanks, looks good!

**Pull request merged**